src/TimeZones.jl

```diff
@@ -0,0 +1,62 @@
+module TimeZones
+
+using Base.Dates
+import Base.Dates: days, hour, minute, second, millisecond
+
+export TimeZone, FixedTimeZone, VariableTimeZone, ZonedDateTime,
+    AmbiguousTimeError, NonExistentTimeError, DateTime,
+    # accessors.jl
+    hour, minute, second, millisecond,
+    # adjusters.jl
+    firstdayofweek, lastdayofweek,
+    firstdayofmonth, lastdayofmonth,
+    firstdayofyear, lastdayofyear,
+    firstdayofquarter, lastdayofquarter,
+    # Re-export from Base.Dates
+    yearmonthday, yearmonth, monthday, year, month, week, day, dayofmonth
+
+const PKG_DIR = normpath(joinpath(dirname(@__FILE__), "..", "deps"))
+const TZDATA_DIR = joinpath(PKG_DIR, "tzdata")
+const COMPILED_DIR = joinpath(PKG_DIR, "compiled")
+
+include("timezones/time.jl")
+include("timezones/types.jl")
+include("timezones/accessors.jl")
+include("timezones/arithmetic.jl")
+include("timezones/io.jl")
+include("timezones/adjusters.jl")
+include("timezones/Olsen.jl")
+
+function TimeZone(name::String)
+    tz_path = joinpath(COMPILED_DIR, split(name, "/")...)
+
+    isfile(tz_path) || error("Unknown timezone $name")
+
+    open(tz_path, "r") do fp
+        return deserialize(fp)
+    end
+end
+
+function timezone_names()
```

Comment:
I was leaning towards calling it `names` and not exporting it. Then the most of calls would be `TimeZones.names()`. Unfortunately `names` lists the functions exported by a module and may be confusing. Calling this function `timezones` could be misleading as someone may expect it to return all the `TimeZone` instances.